src/server/tests.rs

```diff
@@ -144,6 +144,72 @@ async fn test_did_change() {
     assert_eq!(r#"from(bucket: "bucket")"#, contents);
 }
 
+/// When a `textDocument/didChange` presents a file change for a file
+/// using composition, the updated file gets saved on the stateful composition.
+#[test]
+async fn test_did_change_updates_composition_file() {
+    let server = create_server();
+    open_file(&server, "".to_string(), None).await;
+
+    let uri = lsp::Url::parse("file:///home/user/file.flux").unwrap();
+    match server.state.lock() {
+        Ok(mut state) => {
+            let ast = flux::parser::parse_string("".to_string(), "");
+            let composition = composition::Composition::new(
+                ast,
+                "bucket".to_string(),
+                Some("myMeasurement".to_string()),
+                vec![],
+                vec![],
+            );
+            state.set_composition(uri.clone(), composition);
+            let composition_state = state.get_mut_composition(&uri);
+            assert!(composition_state.is_some());
+            assert_eq!(
+                composition_state.unwrap().to_string(),
+                r#"from(bucket: "bucket")
+    |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
+    |> filter(fn: (r) => r._measurement == "myMeasurement")
+"#
+                .to_string(),
+            );
+        }
+        Err(err) => panic!("{}", err),
+    }
+
+    let changed_text = r#"from(bucket: "bucket")
+    |> range(start: -30m)
+    |> filter(fn: (r) => r._measurement == "myMeasurement")
+    |> first()
+"#
+    .to_string();
+    let params = lsp::DidChangeTextDocumentParams {
+        text_document: lsp::VersionedTextDocumentIdentifier {
+            uri: lsp::Url::parse("file:///home/user/file.flux")
+                .unwrap(),
```

Comment:
Looks like this is the same as line 154, just curious why don't you reuse the `uri` from there?

Reply:
Nice catch. Fixing.